src/picongpu/include/plugins/radiation/utilities.hpp

```diff
@@ -50,6 +50,18 @@ namespace util
         return a * a*a;
     }
 
+    template<typename A> /// a generic power-4 function
+    HDINLINE A quad(A a)
+    {
+        return a*a * a*a;
+    }
+
+    template<typename A, typename R> /// a more generic power-4 function
```

Comment:
just added one more to the existing ones
about the general concept and discussions about better ways around it: let's include @PrometheusPi